pkgs/applications/misc/ethereum/cpp/default.nix

```diff
@@ -0,0 +1,134 @@
+{ stdenv
+, fetchgit
+, cmake
+, boost
+, gmp
+, jsoncpp
+, leveldb
+, cryptopp
+, cudatoolkit
+, nvidia_x11
```

Comment:
It might be possible to compile everything with `beignet`. I am going to check it.